python/ray/data/block.py

```diff
@@ -479,3 +479,77 @@ def aggregate_combined_blocks(
     def block_type(self) -> BlockType:
         """Return the block type of this block."""
         raise NotImplementedError
+
+
+def _get_block_boundaries(arrays: list[np.ndarray]) -> np.ndarray:
+    """Compute boundaries of the groups within a block, which is represented
+    by a list of sorted 1D numpy arrays for each column. In each column array,
```

Comment:
This method makes a lofty indirect assumption that arrays are sorted which is only true as long as we're using sorting-based shuffle.

I'd rather instead not assume that and either test or do sorting in it explicitly

Reply:
Actually on a second thought: we actually are not relying on the fact that columns *have to* be sorted inside this method itself. 

Disregard this comment